src/commands/Moderation/purge.ts

```diff
@@ -0,0 +1,36 @@
+import { AstraeaCommand, AstraeaCommandOptions } from '../../lib/Structures/Command'
+import { Message, TextChannel } from 'discord.js'
+import { ApplyOptions, RequiresPermissions } from '@sapphire/decorators'
+import { Args } from '@sapphire/framework'
+
+@ApplyOptions<AstraeaCommandOptions>({
+	name: 'purge',
+	description: 'Makes the bot purge X number of messages',
+	requiredClientPermissions: ['MANAGE_MESSAGES'],
+	cooldownDelay: 2000,
+	cooldownLimit: 2,
+	preconditions: ['GuildTextOnly']
+})
+export default class Purge extends AstraeaCommand {
+	@RequiresPermissions('MANAGE_MESSAGES')
+	// eslint-disable-next-line @typescript-eslint/no-invalid-void-type
+	public async run (message: Message, args: Args): Promise<void | Message> {
+		const amount = (await args.pickResult('number')).value
+
+		if (!amount) return await message.channel.send('You didn\'t provide an amount!')
+		if (amount > 100) return await message.channel.send('I can only purge 100 messages!')
+		if (amount < 1) return await message.channel.send('I can\'t just delete nothing :\\')
+
+		await message.delete()
+
+		const messages = await message.channel.messages.fetch({ limit: amount })
+
+		await (message.channel as TextChannel).bulkDelete(messages.array().slice(0, amount))
```

Comment:
please take into account the message a user sends to initiate this command, which they likely wont have done themselves.